File: presentation/src/main/java/info/firozansari/presentation/view/activity/UserListActivity.java

package info.firozansari.presentation.view.activity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.Window;
import info.firozansari.presentation.R;
import info.firozansari.presentation.internal.di.HasComponent;
import info.firozansari.presentation.internal.di.components.DaggerUserComponent;
import info.firozansari.presentation.internal.di.components.UserComponent;
import info.firozansari.presentation.model.UserModel;
import info.firozansari.presentation.view.fragment.UserListFragment;

/**
 * Activity that shows a list of Users.
 */
public class UserListActivity extends BaseActivity implements HasComponent<UserComponent>,
    UserListFragment.UserListListener {

  public static Intent getCallingIntent(Context context) {
    return new Intent(context, UserListActivity.class);
  }

  private UserComponent userComponent;

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    requestWindowFeature(Window.FEATURE_INDETERMINATE_PROGRESS);
    setContentView(R.layout.activity_layout);

    this.initializeInjector();
    if (savedInstanceState == null) {
      addFragment(R.id.fragmentContainer, new UserListFragment());
    }
  }

  private void initializeInjector() {
    this.userComponent = DaggerUserComponent.builder()
        .applicationComponent(getApplicationComponent())
        .activityModule(getActivityModule())
        .build();
  }

  @Override public UserComponent getComponent() {
    return userComponent;
  }

  @Override public void onUserClicked(UserModel userModel) {
    this.navigator.navigateToUserDetails(this, userModel.getUserId());
  }
}

File: presentation/src/main/java/info/firozansari/presentation/view/activity/UserListActivity.kt
package info.firozansari.presentation.view.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.Window
import info.firozansari.presentation.R
import info.firozansari.presentation.internal.di.HasComponent
import info.firozansari.presentation.internal.di.components.UserComponent
import info.firozansari.presentation.model.UserModel
import info.firozansari.presentation.view.fragment.UserListFragment
import info.firozansari.presentation.view.fragment.UserListFragment.UserListListener

/**
 * Activity that shows a list of Users.
 */
class UserListActivity : BaseActivity(), HasComponent<UserComponent?>, UserListListener {
    private var userComponent: UserComponent? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_INDETERMINATE_PROGRESS)
        setContentView(R.layout.activity_layout)
        initializeInjector()
        if (savedInstanceState == null) {
            addFragment(R.id.fragmentContainer, UserListFragment())
        }
    }

    private fun initializeInjector() {
        userComponent = DaggerUserComponent.builder()
            .applicationComponent(applicationComponent)
            .activityModule(activityModule)
            .build()
    }

    override fun getComponent(): UserComponent? {
        return userComponent
    }

    override fun onUserClicked(userModel: UserModel?) {
        navigator.navigateToUserDetails(this, userModel.getUserId())
    }

    companion object {
        fun getCallingIntent(context: Context?): Intent? {
            return Intent(context, UserListActivity::class.java)
        }
    }
}